modules/rhsm_manifest.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2017, Sean O'Keeffe <seanokeeffe797@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: rhsm_manifest
+short_description: Interact with a Red Hat Satellite Subscription Manifest
+description:
+    - Download and modify a Red Hat Satellite Subscription Manifest
+author:
+    - "Sean O'Keeffe (@sean797)"
+options:
+    name:
+        description:
+           - Manifest Name
+    uuid:
+        description:
+           - Manifest uuid
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    pool_id:
+        description:
+            - Subscription pool_id
+    quantity:
+        description:
+            - quantity of pool_id Subscriptions
+        type: int
+    state:
+        description:
+            - State of the Subscription
+        default: present
+        choices:
+            - present
+            - absent
+    path:
+        description:
+            - path to export the manifest
+        type: path
+    validate_certs:
+        description:
+            - Validate Portal Certs
+        default: True
+        type: bool
+    portal:
+        description:
+            - Red Hat Portal subscription access address
+        default: https://subscription.rhn.redhat.com
+'''
+
+EXAMPLES = '''
+- name: Create katello.example.com Manifest and add 7 sub
+  rhsm_manifest:
+    name: "katello.example.com"
+    username: "john-smith"
+    password: "changeme"
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    quantity: 7
+    state: present
+    validate_certs: false
```

Comment:
If I understand correctly, this example ensures that the manifest would have 7 of pool XXXXX?

Reply:
Correct 👍